pkg/cli/options.go

```diff
@@ -139,3 +150,150 @@ func WithCompletion() Option {
 		return nil
 	}
 }
+
+// parseExternalPluginArgs parses external plugin arguments and trims plugins flag and its values from args.
+func parseExternalPluginArgs() (extPlugins []string, args []string) {
+	var externalPlugins string
+	args = make([]string, len(os.Args)-1)
+	copy(args, os.Args[1:])
+
+	// parse os.Args and pass everything to plugin except program name,
+	// --plugins flag and its value.
+	for i, item := range args {
+		if item == "--plugins" {
+			externalPlugins = args[i+1]
```

Comment:
You can also write multiple plugins like `kubebuilder init --plugins foo/v1 --plugins bar/v2`. To avoid parsing edge cases like this, copy what's going on in [this method](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/cli/cli.go#L235-L271) to extract all plugins from args.

Reply:
Didn't end up using the parsed plugins list so cleaned up that code. Let me know if I still have to follow the same approach from the above link to parse the plugin keys from args. As we are passing unadulterated args to the plugin, figured there's no need to filter out `plugins` flag from args.